### PR TITLE
tpm: Mark the device as updatable if it supports Field Upgrade

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by zaolin** [comment]
Add ReadFirmware commands for verification as well

File: plugins/tpm/fu-tpm-v2-device.c [hunk 1]
```diff
@@ -222,6 +222,55 @@ fu_tpm_v2_device_setup_pcrs(FuTpmV2Device *self, ESYS_CONTEXT *ctx, GError **err
...
+			seen_upgrade_start = TRUE;
+		} else if (cap_cmd == TPM2_CC_FieldUpgradeData) {
+			seen_upgrade_data = TRUE;
+		}
+	}
+	g_debug("CAP_COMMANDS: %s", str->str);
+
+	/* both available */
+	if (seen_upgrade_start && seen_upgrade_data)
+		fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_UPDATABLE);
```
Comment (superm1) [inline plugins/tpm/fu-tpm-v2-device.c:268]: presumably we actually need a routine that can do updates to add this flag

**Comment by hughsie** [comment]
> Add ReadFirmware commands for verification as well

Done.

File: plugins/tpm/fu-tpm-v2-device.c (hunk 1 above)
Comment (hughsie) [reply plugins/tpm/fu-tpm-v2-device.c:268]: Yup, I've just added this although it's 100.00% untested.

**Comment by hughsie** [comment]
So this PR gets us 70% the way there -- the vendors would still need to tell us how to set up the transfer. I can't find any vendor that does TPM updates using the standardized API, and so this code is more of a "this is how it should be implemented" rather than with any expectation it's going to work. Unless that is the vendor allows flashing unsigned firmware, which for a TPM would be **bad**.

**Comment by hughsie** [comment]
> so this code is more of a "this is how it should be implemented" rather

It's also to show Infineon how to do it with their vendor update flow...

**Pull request merged**